src/main/java/com/google/ehub/utility/UserRecommendationUtils.java

```diff
@@ -15,18 +16,55 @@
 
   /**
    * Finds the most recommended emails up to a maximum of ten recommendations in descending order.
+   * In the case of a tie, the order will use increasing lexicographical order.
    *
    * @param itemLikes map with the key representing an itemId and the value representing the emails
    *     that liked that item
    * @return list containing the most recommended emails in descending order
    */
   public List<String> getRecommendedEmails(Map<Long, Set<String>> itemLikes) {
+    return getRecommendedEmailsInDescendingOrder(getEmailPriorityQueue(getEmailFreqs(itemLikes)));
```

Comment:
this is getting a little hard to read (chaining 3 method calls). it might be better to just split them into some local variables

Reply:
OK I remove the chain of calls by adding local variables in last commit.